### PR TITLE
Remove incorrect signal error stack trace

### PR DESCRIPTION
**Pull request by chenwhql**
### PR types

Bug fixes
### PR changes

Others
### Describe

The original signal error stacktrace like follow

```
--------------------------------------
C++ Traceback (most recent call last):
--------------------------------------
0   paddle::framework::SignalHandle(char const*, int)
1   paddle::platform::GetCurrentTraceBackString[abi:cxx11]()

----------------------
Error Message Summary:
----------------------
FatalError: `Segmentation fault` is detected by the operating system.
  [TimeInfo: *** Aborted at 1628745164 (unix time) try "date -d @1628745164" if you are using GNU date ***]
  [SignalInfo: *** SIGSEGV (@0x15e6) received by PID 5606 (TID 0x7fe7d4604700) from PID 5606 ***]

[1]    5606 segmentation fault  python sig_segv.py
```

The trace of `SignalHandle` and `GetCurrentTraceBackString` are not the signal error stack, which are introduced by getting traceback, this may misleading users, so this PR remove this two line.

And if no paddle stack trace when signal error occurred, we also give users hint. For this case, the new result like:

```
--------------------------------------
C++ Traceback (most recent call last):
--------------------------------------
No stack trace in paddle, may be caused by external reasons.

----------------------
Error Message Summary:
----------------------
FatalError: `Segmentation fault` is detected by the operating system.
  [TimeInfo: *** Aborted at 1628743780 (unix time) try "date -d @1628743780" if you are using GNU date ***]
  [SignalInfo: *** SIGSEGV (@0xaa6) received by PID 2726 (TID 0x7f599fa2f700) from PID 2726 ***]

[1]    2726 segmentation fault  python sig_segv.py
```

### CODE REVIEW

File: paddle/fluid/platform/enforce.h [hunk 1]
```diff
@@ -284,7 +286,12 @@ inline std::string GetCurrentTraceBackString() {
   auto symbols = backtrace_symbols(call_stack, size);
   Dl_info info;
   int idx = 0;
-  for (int i = size - 1; i >= 0; --i) {
+  // `for_signal` used to remove the stack trace introduced by
+  // obtaining the error stack trace when the signal error occurred,
+  // that is not related to the signal error self, remove it to
+  // avoid misleading users and developers
+  int end_idx = 0 ? for_signal : 2;
```
Comment (MingMingShangTian) [inline paddle/fluid/platform/enforce.h]: 这里应该是 for_signal ? 0 : 2;    吧？

**Review by hbwx24 — DISMISSED** [review]
LGTM

File: paddle/fluid/platform/enforce.h (hunk 1 above)
Comment (jim19930609) [inline paddle/fluid/platform/enforce.h]: should it be "for_signal ? 2 : 0" ?

Comment (chenwhql) [reply paddle/fluid/platform/enforce.h]: done, thx, `for_signal ? 2 : 0`

Comment (chenwhql) [reply paddle/fluid/platform/enforce.h]: done, thx

**Review by jim19930609 — APPROVED** [review]

**Review by MingMingShangTian — APPROVED** [review]
LGTM

**Pull request merged**